### PR TITLE
[Federation] Improve e2e test setup

### PR DESCRIPTION
**Pull request by marun**
This PR improves federation e2e test setup:

 - reuses e2e framework setup (``NewDefaultFramework``) instead of duplicating it
 - ensures ``FederationAfterEach`` is called if an error occurs in ``FederationBeforeEach`` (as per the [example](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L161) of the e2e framework)
 - skips creation of a test namespace in the hosting cluster (not used for a federation e2e test)

cc: @kubernetes/sig-federation-pr-reviews @kubernetes/sig-testing-pr-reviews

### CODE REVIEW

File: test/e2e_federation/framework/framework.go [hunk 1]
```diff
@@ -40,33 +40,40 @@ import (
...
+	// framework's cleanup.
+	AfterEach(f.FederationAfterEach)
 
-	f := &Framework{&framework.Framework{
-		BaseName:                 baseName,
-		AddonResourceConstraints: make(map[string]framework.ResourceConstraint),
-		Options:                  options,
-		ClientSet:                nil,
-	}, nil, &v1.Namespace{}}
+	f.Framework = framework.NewDefaultFramework(baseName)
```
Comment (jayunit100) [inline test/e2e_federation/framework/framework.go]: ugh why does &Framework{} have a Framework ? weird . but i think it makes sense.

File: test/e2e_federation/framework/framework.go [hunk 2]
```diff
@@ -40,33 +40,40 @@ import (
...
-	options := framework.FrameworkOptions{
-		ClientQPS:   20,
-		ClientBurst: 50,
-	}
+	f := &Framework{}
+
+	// Register the federation cleanup before initializing the default
+	// e2e framework to ensure it gets called before the default
+	// framework's cleanup.
+	AfterEach(f.FederationAfterEach)
```
Comment (jayunit100) [inline test/e2e_federation/framework/framework.go]: ahhh clever :)

**Comment by shashidharatd** [comment]
This is cool, good optimization. LGTM.
although would wait for federation pre-commit tests to succeed.

**Comment by madhusudancs** [comment]
/assign @madhusudancs

---

Reviewed 1 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44066#-:-KgzjpfDRKLi-3WeBfn8:b59i52z)*

**Comment by madhusudancs** [comment]
/lgtm

---

Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44066#-:-Kgzjsuk9V5qTGk4yfyo:b-8ks5qa)*

**Comment by madhusudancs** [comment]
Reviewed 1 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/44066)*

**Pull request merged**